src/main/java/com/learn/hashmap/OrderPlacingSystem.java

```diff
@@ -0,0 +1,80 @@
+package com.learn.hashmap;
+
+import java.util.*;
+
+/**
+ * Advanced Real-World Problem: Order Fulfillment System
+ * Problem Statement:
+ * You are tasked with implementing an Order Fulfillment System for an e-commerce platform.
+ * The system processes incoming orders and maintains a record of customer orders using a HashMap.
+ * Each customer can have multiple orders stored as a list. The system should support the following operations:
+ * <p>
+ * 1. Add a new order for a customer: If the customer already exists, add the order to their list.
+ * If the customer does not exist, create a new entry with the customer's ID and their order.
+ * <p>
+ * 2. Retrieve all orders for a customer: Retrieve the list of orders for a specific customer.
+ * <p>
+ * 3. Check if a customer exists: Check if a specific customer ID is already in the system.
+ * <p>
+ * 4. Add a default order for a customer if absent: If the customer ID does not exist, add a default order (e.g., "Welcome Package").
+ * <p>
+ * 5. Update the last order's status: For a given customer ID, update the status of their most recent order. If no order exists, do nothing.
+ */
+public class OrderPlacingSystem {
+
+    private Map<String, List<Order>> ordersMap;
+
+    public OrderPlacingSystem() {
+        ordersMap = new HashMap<>();
+    }
+
+    public void addOrderForCustomer(String customerId, Order order) {
```

Comment:
There are map methods which can do this more efficiently and concisely. The purpose of this exercise to learn map.